.circleci/config.yml

```diff
@@ -97,12 +97,90 @@ jobs:
             coverage run --source=super_gradients -m unittest tests/deci_core_unit_test_suite_runner.py
             coverage report
             coverage html  # open htmlcov/index.html in a browser
-
       - store_artifacts:
           path: htmlcov
 
       - store_artifacts:
           path: ~/sg_logs
+      
+  docker_build:
+    parameters:
+        version:
+          type: string
+          default: latest
+    environment: 
+      IMAGE_TAG: << parameters.version >>
+    description: "building the docker"
+    machine: true 
+    resource_class: deci-ai/research-standard
+    steps:
+      - checkout
+      - run:
+         name: making sure we have docker permmissions #ugly but working solution
+         command: sudo chmod 666 /var/run/docker.sock
+      - deci-common/build_docker_image:
+          image_tag: << parameters.version >>
+          build_args: VERSION=${CIRCLE_BRANCH}
+          dockerfile: "scripts/Dockerfile"
+          repository_name: "deciai/super-gradients"
+      - run:
+          name: Archive Docker image
+          command: docker save -o image.tar deciai/super-gradients:${IMAGE_TAG} 
+      - persist_to_workspace:
```

Comment:
This can cost a lot of money on a self hosted runenr

Reply:
i only persisting 
as i read from circleci website 

The only network traffic that will result in billing is accrued through restoring caches and workspaces to self-hosted runners. Retention of artifact, workspace, and cache objects will result in billing for storage usage.

so it should be free , i can also build the docker in the cloud and run in on prem but this complicates and add download cost and ecr cost for temporary images